app/waf/src/controllers/rule.controller.ts

```diff
@@ -139,4 +147,239 @@ export class RuleController {
   async deleteById(@param.path.string('id') id: string): Promise<void> {
     await this.ruleRepository.deleteById(id);
   }
+  /*Move the conditions and actions related functions here.*/
+
+  @get(prefix + '/actions/count', {
+    responses: {
+      '200': {
+        description: 'Action model count',
+        content: {'application/json': {schema: CountSchema}},
+      },
+    },
+  })
+  async countActions(
+    @param.query.object('where', getWhereSchemaFor(Action)) where?: Where,
+  ): Promise<Count> {
+    return await this.actionRepository.count(where);
+  }
+
+  @get(prefix + '/actions', {
+    responses: {
+      '200': {
+        description: 'Array of Action model instances',
+        content: {
+          'application/json': {
+            schema: {type: 'array', items: {'x-ts-type': Action}},
+          },
+        },
+      },
+    },
+  })
+  async findActions(
+    @param.query.object('filter', getFilterSchemaFor(Action)) filter?: Filter,
+  ): Promise<Action[]> {
+    return await this.actionRepository.find(filter);
+  }
+
+  @get(prefix + '/conditions/count', {
+    responses: {
+      '200': {
+        description: 'Condition model count',
+        content: {'application/json': {schema: CountSchema}},
+      },
+    },
+  })
+  async countConditions(
+    @param.query.object('where', getWhereSchemaFor(Condition)) where?: Where,
+  ): Promise<Count> {
+    return await this.conditionRepository.count(where);
+  }
+
+  @get(prefix + '/conditions', {
+    responses: {
+      '200': {
+        description: 'Array of Condition model instances',
+        content: {
+          'application/json': {
+            schema: {type: 'array', items: {'x-ts-type': Condition}},
+          },
+        },
+      },
+    },
+  })
+  async findConditions(
+    @param.query.object('filter', getFilterSchemaFor(Condition))
+    filter?: Filter,
+  ): Promise<Condition[]> {
+    return await this.conditionRepository.find(filter);
+  }
+
+  @post(prefix + '/rules/{rule_id}/conditions', {
+    responses: {
+      '200': {
+        description: 'Conditions add to rule success',
+        content: {'application/json': {schema: {'x-ts-type': Condition}}},
+      },
+    },
+  })
+  async createRuleCondition(
+    @param.path.string('rule_id') rule_id: string,
+    @requestBody() condition: Partial<Condition>,
+  ): Promise<Condition> {
+    return await this.ruleRepository.conditions(rule_id).create(condition);
+  }
+
+  @get(prefix + '/rules/{rule_id}/conditions/{condition_id}', {
+    responses: {
+      '200': {
+        description: 'Condition model instance',
+        content: {'application/json': {schema: {'x-ts-type': Condition}}},
+      },
+    },
+  })
+  async getConditionByID(
+    @param.path.string('rule_id') rule_id: string,
+    @param.path.string('condition_id') condition_id: string,
+  ): Promise<Condition> {
+    const result = await this.ruleRepository
```

Comment:
Here, it's suitable to use findById instead.

Reply:
resolved it.